extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -36,57 +35,100 @@ public DevConsoleTemplateInfoBuildItem collectAdditionalEndpoints(
+                            List<StaticResourceInfo.StaticFile> childList = new ArrayList<>();
+                            childList.add(newStaticFile);
+                            StaticResourceInfo.StaticFile parentStaticFile = new StaticResourceInfo.StaticFile(
+                                    parentOfDir.toString(), true, childList);
+                            staticResourceInfo.resourceMap.put(parentOfDir, parentStaticFile);
                         }
                     }
-                    if (!file.startsWith("/")) {
-                        file = "/" + file;
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attributes)
+                        throws IOException {
+                    String fileName = file.getFileName().toString();
+                    String completeParentPath = file.getParent().toString();
+                    String relativeFileName = resource.relativize(file).toString();
+                    Path parentDirectory = file.getParent().getFileName();
+
+                    StaticResourceInfo.StaticFile childStaticFile = new StaticResourceInfo.StaticFile(fileName,
+                            attributes.isDirectory(), null);
+
+                    // Add child static in Parent Folder
+                    if (staticResourceInfo.resourceMap.containsKey(parentDirectory)) {
+                        StaticResourceInfo.StaticFile valueOfParentKey = staticResourceInfo.resourceMap.get(parentDirectory);
+                        List<StaticResourceInfo.StaticFile> retrieveChildList = valueOfParentKey.children;
+                        retrieveChildList.add(childStaticFile);
+                        StaticResourceInfo.StaticFile updatedParent = new StaticResourceInfo.StaticFile(
+                                parentDirectory.toString(), true, retrieveChildList);
+                        staticResourceInfo.resourceMap.put(parentDirectory, updatedParent);
+                    } else {
+                        List<StaticResourceInfo.StaticFile> childList = new ArrayList<>();
+                        childList.add(childStaticFile);
+                        StaticResourceInfo.StaticFile parentStaticFile = new StaticResourceInfo.StaticFile(
+                                parentDirectory.toString(), true, childList);
+                        staticResourceInfo.resourceMap.put(parentDirectory, parentStaticFile);
                     }
-                    file = file.replace('\\', '/');
-                    staticResources.add(file);
+
+                    return FileVisitResult.CONTINUE;
+                }
+
+                //in post-visit you need to return back to the previous folder you were visiting.
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
```

Comment:
It doesn't look like you've figured out the right data structure to keep track of the current folder, so I'll help you: it's a stack ;)

Reply:
In this part of "keeping track of current folder", I have some doubts.
Do we simply need to create a stack `currentFolder` in `collectKnownPaths()`, add current folder into the stack in `previsit` then pop that in `postvisit` to return to previous folder? 
Probably I am wrong here 🤔 

"We want to keep collection of all folders we are visiting like Resources, Images, Javascript"

Reply:
> Do we simply need to create a stack currentFolder in collectKnownPaths(), add current folder into the stack in previsit then pop that in postvisit to return to previous folder?

Yes, but you can keep it as a field in your visitor. If you use a stack then the top of the stack is always the current folder that you need to add children to.

Reply:
Ok :)